### ISSUE TITLE
Research: Fix estimates on results for Timeline Visualization/List

### ISSUE THREAD

**Issue by roamye** [project-lux/lux-marklogic#406]
**Problem Description:**
The results shown on the timeline visualization and list view display incorrect results on it's result pages. 

**Expected Behavior/Solution:**
The results listed within the timeline visualization and list view must match the results listed on the search result page. 

**Requirements:**
- Result links on the timeline visualization and list view must match the number that displays on its query results. 
- TBD based on research

**Needed for promotion:**
If an item on the list is not needed, it should be crossed off but not removed.

- [ ] Wireframe/Mockup - Mike
- [ ] Committee discussions - Sarah
- [ ] Feasibility/Team discussion - Sarah
- [ ] Backend requirements - TBD
- [ ] Frontend requirements- TBD
- [ ] Are new regression tests required for QA - Amy
- [ ] Questions
- List of questions for discussions. Answers should be documented within the issue.

**UAT/LUX Examples:**
- https://lux-front-tst.collections.yale.edu/view/person/34f4eec7-7a03-49c8-b1be-976c2f6ba6ba

**Dependencies/Blocks:**
Blocked by: Research

**Related Github Issues:**
N/A 

**Related links:**
- Bugherd: https://www.bugherd.com/projects/284041/tasks/2755

**Wireframe/Mockup:**
Place wireframe/mockup for the proposed solution at end of ticket.

**Comment by jffcamp** [comment]
Discussed during 7/2 ticket forming. Deferred until after My Collections MVP is launched.

**Comment by zachjschwartz** [comment]
TF 4/21 (discussing https://www.bugherd.com/projects/284041/tasks/3223): Clear to un-defer. Peter will look into it. Will require some strategizing.

**Comment by clarkepeterf** [comment]
I added this to priortization review

**Comment by zachjschwartz** [comment]
@brent-hartwig is this worth researching prior to the change to Optic? Or should this be part of the Optic work, or done after?

**Comment by brent-hartwig** [comment]
@zachjschwartz, I'm not familiar with this ticket but if the timeline is dependent on estimates from MarkLogic, let's wait until after the Optic migration.

**Comment by clarkepeterf** [comment]
@zachjschwartz @brent-hartwig I think it is worth considering while Optic is being worked on.

The solution is either to:
1. Have facets counts for production/creation dates that **overlap** with the year in question
OR
2. Search for creation/production specific to the year in question

Example of the issue:
[Andy Warhol](https://lux-front-tst.collections.yale.edu/view/person/66049111-383e-4526-9632-2e9b6b6302dd)'s Timeline says 2 objects produced in 1965. This is because out of a search for objects created by Andy Warhol, 2 of them have a itemProductionStartDate in 1965.

If you click the link, you get [a search](https://lux-front-tst.collections.yale.edu/view/results/objects?q=%7B%22AND%22%3A%5B%7B%22producedBy%22%3A%7B%22id%22%3A%22https%3A%2F%2Flux.collections.yale.edu%2Fdata%2Fperson%2F66049111-383e-4526-9632-2e9b6b6302dd%22%7D%7D%2C%7B%22AND%22%3A%5B%7B%22producedDate%22%3A%221965-01-01T00%3A00%3A00.000Z%22%2C%22_comp%22%3A%22%3E%3D%22%7D%2C%7B%22producedDate%22%3A%221965-01-01T00%3A00%3A00.000Z%22%2C%22_comp%22%3A%22%3C%3D%22%7D%5D%7D%5D%7D&collapseSearch=true) that will return any objects with creation dates that **overlap** with 1965.

Many of the objects have an ambiguous creation date like this:

```
"end_of_the_end": "2000-12-31T00:00:00Z",
"begin_of_the_begin": "1900-01-01T00:00:00Z",
"_seconds_since_epoch_begin_of_the_begin": -2208988800,
"_seconds_since_epoch_end_of_the_end": 978220800
```

All of these **overlap** with 1965, so they match the search and add to the total count. Where, the facet ignored overlapping dates.

**Comment by clarkepeterf** [comment]
@zachjschwartz my other question would be, what are the user expectations for this?

If an object's data says it was created "Between 1900 and 2000", do we want the timeline to include it as an Object Created in the year 1900, the year 2000, or for every year in between?

**Comment by brent-hartwig** [comment]
@jffcamp and @prowns, Peter makes a valid point that we may want to take this need into account while working on Optic vs. afterwards; however, it's not scope I was tracking.  Please advise.

**Comment by prowns** [comment]
@brent-hartwig  - please consider this with Optic.

**Comment by clarkepeterf** [comment]
@jffcamp @prowns  If an object's data says it was created "Between 1900 and 2000", do we want the timeline to include it as an Object Created in the year 1900, the year 2000, or for every year in between?

If we use my proposed solution 2 [from above](https://github.com/project-lux/lux-marklogic/issues/406#issuecomment-4347022700), this is functionality we already agreed on for the Optic date pattern. But I want to make sure our solution matches user requirements.
cc @brent-hartwig

**Comment by prowns** [comment]
@clarkepeterf  - to confirm - Using your example for proposal 2, the system will only return the 1965 objects from the time timeline --> results, so the estimate and actual results match.  Is this correct?

**Comment by clarkepeterf** [comment]
@prowns with either solution 1 or 2, the timeline and results will match.

With 1: The timeline will show 29 objects in 1965, and results will show 29
With 2: The timeline will show 2 objects in 1965, and results will show 2

**Comment by brent-hartwig** [comment]
I suggest interpreting "Between 1900 and 2000" as `>= 1900-01-01T00:00:00` and `<= 2000-12-31T23:59:59` (option 1).

**Comment by clarkepeterf** [comment]
@brent-hartwig so you'd like to change the timeline such that this would add +1 to the count for every year between 1900 and 2000?

```
"end_of_the_end": "2000-12-31T00:00:00Z",
"begin_of_the_begin": "1900-01-01T00:00:00Z",
"_seconds_since_epoch_begin_of_the_begin": -2208988800,
"_seconds_since_epoch_end_of_the_end": 978220800
```

The tricky part about option 1 is that this doesn't map onto the way facets operate currently (get the value of an index for every doc, then count), rather it looks something like:
1. find earliest year
2. find latest year
3. for every year from earliest to latest, calculate a count of how many docs have overlapping date ranges

**Comment by prowns** [comment]
@clarkepeterf - Decision: Result for single year on the timeline should get results for that specific year only  (from 5/1 team mtg)

- will need new Middle Tier ticket, and perhaps front end.  TBD if optic affects the ML layer. 
- Showing the ambiguous date results set will still be a disconnect.  Eg. ambiguous results will show on timeline, but results will be wrong 

@azaroth42 , @brent-hartwig @clarkepeterf  - to work on ambiguous date problem.

**Comment by brent-hartwig** [comment]
@clarkepeterf and @kamerynB, as part of or following the Optic migration, the frontend will need to change its search to a single date search term set to equality and specifying the "begin" timeline mode.  E.g.,:

```json
{
  "startDate":"2012",
  "_comp":"=",
  "_timespanMode":"begin"
}
```

**Comment by kamerynB** [comment]
@brent-hartwig @clarkepeterf Thank you Brent. I don't think the frontend requires any change for timelines specifically. I think the change happens in the MT or ML. The frontend is just readying the data for the timelines.
